sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/DirectoryAPITest.groovy

```diff
@@ -315,21 +492,18 @@ class DirectoryAPITest extends APISpec {
         }
         def options = new DataLakePathCreateOptions().setMetadata(metadata)
 
-        when:
-        dc.createIfNotExistsWithResponse(options, null, Context.NONE)
-        def response = dc.getProperties()
-
-        then:
+        expect:
+        dc.createIfNotExistsWithResponse(options, null, null).getStatusCode() == statusCode
         // Directory adds a directory metadata value
         for(String k : metadata.keySet()) {
-            response.getMetadata().containsKey(k)
-            response.getMetadata().get(k) == metadata.get(k)
+            dc.getProperties().getMetadata().containsKey(k)
+            dc.getProperties().getMetadata().get(k) == metadata.get(k)
         }
 
         where:
-        key1  | value1 | key2   | value2
-        null  | null   | null   | null
-        "foo" | "bar"  | "fizz" | "buzz"
+        key1  | value1 | key2   | value2 || statusCode
+        null  | null   | null   | null   || 201
```

Comment:
Since the status code is the same for all cases, it doesn't actually need to be in the table--it should just be a constant in the test